File: Server/models/AIPromptLog.js
const mongoose = require('mongoose');
const Counter = require('./Counter');

const aiPromptLogSchema = new mongoose.Schema({
  ailog_id: {
    type: Number,
    unique: true
  },
  session_id: {
    type: String,
    required: true,
    ref: 'AISession',
    index: true
  },
  input_prompt: {
    type: String,
    required: true 
  },
  input_image_url: {
    type: String,
    default: null
  },
  gemini_raw_response: {
    type: String,
    default: null
  },
  weather_context: {
    type: String,
    default: null 
  },
  created_at: {
    type: Date,
    default: Date.now
  }
});

aiPromptLogSchema.index({ created_at: -1 });
aiPromptLogSchema.pre('save', async function() {
  const doc = this;

  if (!doc.isNew) {
    return;
  }

  const counter = await Counter.findByIdAndUpdate(
    { _id: 'ailog_id' },
    { $inc: { seq: 1 } },
    { new: true, upsert: true }
  );

  doc.ailog_id = counter.seq;
});

module.exports = mongoose.model('AIPromptLog', aiPromptLogSchema, 'ai_prompt_logs');

File: Android/app/src/main/java/com/example/smartfashion/model/AIPromptLog.kt
package com.example.smartfashion.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class AIPromptLog(
    @SerializedName("ailog_id")
    val aiLogId: Int? = null,

    @SerializedName("session_id")
    val sessionId: String,

    @SerializedName("input_prompt")
    val inputPrompt: String,

    @SerializedName("input_image_url")
    val inputImageUrl: String? = null,

    @SerializedName("gemini_raw_response")
    val geminiRawResponse: String? = null,

    @SerializedName("weather_context")
    val weatherContext: String? = null,

    @SerializedName("created_at")
    val createdAt: Date? = null
)